app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -23,23 +23,18 @@ public void simpleCheck() throws Exception {
                 "const uint16_t CAN_ID2 = 0x32;\n" +
                 "const uint16_t CAN_ID3 = 0x64;";
         Parse test = new Parse(code.toCharArray());
-        HashMap<String, ConstContents> repository = test.constRepo;
 
-        assertEquals(true, repository.containsKey("CAN_ID"));
-        assertEquals(true, repository.containsKey("CAN_ID2"));
-        assertEquals(true, repository.containsKey("CAN_ID3"));
+        assertEquals("const", test.getConstContents("CAN_ID").typeQualifer);
+        assertEquals("const", test.getConstContents("CAN_ID2").typeQualifer);
+        assertEquals("const", test.getConstContents("CAN_ID3").typeQualifer);
 
-        assertEquals("const", test.getConst("CAN_ID").typeQualifer);
-        assertEquals("const", test.getConst("CAN_ID2").typeQualifer);
-        assertEquals("const", test.getConst("CAN_ID3").typeQualifer);
+        assertEquals("uint16_t", test.getConstContents("CAN_ID").payLoadDataType);
+        assertEquals("uint16_t", test.getConstContents("CAN_ID2").payLoadDataType);
+        assertEquals("uint16_t", test.getConstContents("CAN_ID3").payLoadDataType);
 
-        assertEquals("uint16_t", test.getConst("CAN_ID").type);
-        assertEquals("uint16_t", test.getConst("CAN_ID2").type);
-        assertEquals("uint16_t", test.getConst("CAN_ID3").type);
-
-        assertEquals("0x16", test.getConst("CAN_ID").value);
-        assertEquals("0x32", test.getConst("CAN_ID2").value);
-        assertEquals("0x64", test.getConst("CAN_ID3").value);
+        assertEquals("0x16", test.getConstContents("CAN_ID").value);
+        assertEquals("0x32", test.getConstContents("CAN_ID2").value);
+        assertEquals("0x64", test.getConstContents("CAN_ID3").value);
```

Comment:
Can you also add a test somewhere for if we try to decode a CAN ID that doesn't exist? We do not have all the CAN IDs defined in can_struct.h, and in any case the parser should be robust to when data is unavailable.